rl_coach/env.sh

```diff
@@ -9,7 +9,27 @@ export AWS_DEFAULT_REGION=us-east-1
 export MODEL_S3_PREFIX=rl-deepracer-sagemaker
 export MODEL_S3_BUCKET=bucket
 export LOCAL=True
-export S3_ENDPOINT_URL=http://$(hostname -i):9000
+
+# Check if the "hostname -i" command works
+if hostname -i 2>/dev/null ;
+then
+  export S3_ENDPOINT_URL=http://$(hostname -i):9000  
+else  
+  # On Macs hostname doesn't support a -i option
+  IP_ADDRESS=`ifconfig | grep "inet " | grep -m 1 -Fv 127.0.0.1 | awk '{print $2}'`
+  export S3_ENDPOINT_URL=http://${IP_ADDRESS}:9000
+fi
+
 export MARKOV_PRESET_FILE=deepracer.py
-export LOCAL_ENV_VAR_JSON_PATH=$(readlink -f ./env_vars.json)
+
+# Check if the "readlink -f" command works
+if readlink -f ./env_vars.json 2>/dev/null ;
```

Comment:
Could probably change this to `if [[ -X $(which readlink) ]] ;` to test if the file really exists, and is executable, rather than invoking it and assuming the output will be testable. You could alse change the else to an else if and check the existence of greadlink as well

Reply:
We do need to check `readlink -f` because Mac 10.14.6 has `readlink`, it just doesn't support the `-f` option:
```
$ which readlink
/usr/bin/readlink

$ readlink -f
readlink: illegal option -- f
usage: readlink [-n] [file ...]
```
But I added the test you suggested, checked for `greadlink` too, and added error handling.